### PR TITLE
Refine undo implementation

### PR DESCRIPTION
**Pull request by turtle96**
created History interface and HistoryManager that implements that interface
changed "list-undo" to "history"/"hist" to prevent confusion of listing in which panels (all list commands is for left panel, and calendar and history is for right panel)

also implemented "undo" command for undoing most recent command, to prevent having to type "hist" then "undo INDEX"

### CODE REVIEW

**Comment by turtle96** [comment]
@CS2103AUG2016-W15-C3/testing 
@CS2103AUG2016-W15-C3/documentation 
@CS2103AUG2016-W15-C3/code-quality 

Please review

**Comment by ZuoZhuolin** [comment]
Is it possible to update the command history on the right panel once a command is executed?
Test reviewed.

File: docs/UserGuide.md [hunk 1]
```diff
@@ -138,12 +138,30 @@ Diagram 3 and 4: Keying in `find report` displays list of tasks with report as o
 
 <!--- @@author A0142130A ---> 
 
+#### Showing history : `history` or `hist`
+You can view the command history available for undo on right panel to refer to when undoing previous commands.<br>
```
Comment (Mamtha3005) [inline docs/UserGuide.md]: "on the right panel" --> include the word the

File: docs/UserGuide.md [hunk 2]
```diff
@@ -138,12 +138,30 @@ Diagram 3 and 4: Keying in `find report` displays list of tasks with report as o
...
+To save your time, Taskell has a short form command `hist`. <br>
```
Comment (Mamtha3005) [inline docs/UserGuide.md:143]: This is a callout.

**Comment by Mamtha3005** [comment]
Documentation Reviewed.

File: src/main/java/seedu/taskell/ui/DisplayPanel.java [hunk 3]
```diff
@@ -16,9 +20,18 @@
  */
 public class DisplayPanel extends UiPart{
```
Comment (marcusngwenjian) [inline src/main/java/seedu/taskell/ui/DisplayPanel.java]: Whitespace needed
image

**Comment by turtle96** [comment]
@ZuoZhuolin it is possible but the default display is meant to be calendar view

File: docs/UserGuide.md (hunk 2 above)
Comment (turtle96) [reply docs/UserGuide.md:143]: Do you mean this whole section needs to be a callout?

**Comment by marcusngwenjian** [comment]
@turtle96 
I believe he meant state-dependent behaviour
image
Figure 1. Lecture 10, 36:13

That means to say, while the right panel is still showing the history, it can automatically updates the history w/o user consistently writing hist. However, if the view on the right pane is a calendar, it will not swap back to history and updates it.

**Comment by turtle96** [comment]
@marcusngwj @ZuoZhuolin 
the general view now is calendar, then if "hist" is executed then it shows history, then for the next commands executed it will revert back to calendar
I am aware of the concept of state-dependent behaviour, however, currently the user is unlikely to keep undoing stuff again and again at the same time, so reverting view back to calendar is recommended, it is also recommended by the tutor

calendar is not permanent view at the moment, since it is still undergoing construction, especially for task view support. Also this branch is for undo command so it would be inappropriate to mix in calendar view

hope this clarifies everything

**Comment by marcusngwenjian** [comment]
Understood.
@CS2103AUG2016-W15-C3/integration Ready for integration

**Pull request merged**